Automattic/wp-calypso
# Automate the testing of popular blocks in current and edge versions of GB

#### Changes proposed in this Pull Request

* Automates steps 1 to 7 of the Good Mountain testing checklist but does not focus on testing individual blocks (this will come in subsequent PR(s));
* Look for blocks on the page and also look for block errors on the page;
* Takes screenshots from edge and non-edge editor and preview.
* Do that for edge/non-edge sites across the most popular themes
* Adds a new test account and several new test sites (the account credentials have been added to the encrypted config file)

Repeat that for the most popular themes in wpcom.


#### Testing instructions

1. Setup your env to run Calypso E2E tests (search for "Automated end-to-end Testing") in the FG;
1. Run with `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-upgrade-spec.js`*[0]
1. Wait for the tests to finish. It might take a good while since it tests everything for each test site
1. Make sure it didn't err, if it did, it might not be an actual failure, try running that specific step again. If it continues failing, let us know
1. Look inside the screenshot*[1] folder for the screenshots. There should be a directory per test site name. You can find the list of test sites in the [test file](https://github.com/Automattic/wp-calypso/pull/45048/files#diff-87bdd0c59221389b9b5d297efbdb1b31R117) itself. They are all part of the same test account called `gutenbergUpgradeUser`. You can verify it (alongside the credentials) by decrypting the config file in this PR and looking for it in your `config/local-decrypted.json` file.

*[0] _Make sure to cd into `test/e2e` first._
*[1] `test/e2e/screenshots/`



#### TODO

- [x] Figure out if setting up blocks in a wpcom site then copying the block markup to another renders the blocks correctly in the new site (it looks like it does! This means we can postpone for now the idea of switching gutenberg or applying stickers using a query param)
- [x] Populate tiled-gallery with sample images
- [x] Take screenshots of the editor and preview in the non-edge site
- [x] Take screenshots of the editor and preview in the edge site, after loading the blocks markup from non-edge
- [x] Research how to set it up as a scheduled test and to run it using the "Needs e2e testing Gutenberg edge" tag (we'll just let it run as part of the current e2e pipeline)
